Minio/ApiEndpoints/BucketOperations.cs

```diff
@@ -441,5 +448,130 @@ public IObservable<MinioNotificationRaw> ListenBucketNotificationsAsync(string b
+                SetTargetURL(requestUrl);
+                VersioningConfiguration config = new VersioningConfiguration(true);
+                var request = new RestRequest("/" + bucketName, Method.PUT)
+                {
+                    XmlSerializer = new RestSharp.Serializers.DotNetXmlSerializer()
+                    {
+                        Namespace = "http://s3.amazonaws.com/doc/2006-03-01/",
+                        ContentType = config.ToString()
+                    },
+                    RequestFormat = DataFormat.Xml
+                };
+                string body = Regex.Replace(request.XmlSerializer.Serialize(config), @"\t|\n|\r", "");
+                string replace1 = ">\\s+";
+                string replace2 = "\\s+<";
+                body = Regex.Replace(body, replace1, ">");
+                body = Regex.Replace(body, replace2, "<");
+                request.AddQueryParameter("versioning","");
+                request.AddParameter("text/xml", body, ParameterType.RequestBody);
+                var response = await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+        }
+        /// <summary>
+        /// Enable versioning on a bucket
+        /// </summary>
+        /// <param name="bucketName">Name of the new bucket</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
+        /// <returns>Task that returns true if versioning is enabled on the bucket</returns>
+        public async Task DisableVersioningAsync(string bucketName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            try
+            {
+                if (bucketName == null)
+                {
+                    throw new InvalidBucketNameException(bucketName, "bucketName cannot be null");
+                }
+                Uri requestUrl = RequestUtil.MakeTargetURL(this.BaseUrl, this.Secure);
+                SetTargetURL(requestUrl);
+                VersioningConfiguration config = new VersioningConfiguration(false);
+                var request = new RestRequest("/" + bucketName, Method.PUT)
+                {
+                    XmlSerializer = new RestSharp.Serializers.DotNetXmlSerializer()
+                    {
+                        Namespace = "http://s3.amazonaws.com/doc/2006-03-01/",
+                        ContentType = config.ToString()
+                    },
+                    RequestFormat = DataFormat.Xml
+                };
+                string body = Regex.Replace(request.XmlSerializer.Serialize(config), @"\t|\n|\r", "");
+                string replace1 = ">\\s+";
+                string replace2 = "\\s+<";
+                body = Regex.Replace(body, replace1, ">");
+                body = Regex.Replace(body, replace2, "<");
```

Comment:
same comment as above - xml needs to be marshalled and UTF-8 encoded.

Reply:
Made the change here too.